cpp/CMakeLists.txt

```diff
@@ -14,169 +14,66 @@
-if(CMAKE_CUDA_LINEINFO)
-    set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -lineinfo")
-endif(CMAKE_CUDA_LINEINFO)
-
-# Debug options
-if(CMAKE_BUILD_TYPE MATCHES Debug)
-    message(STATUS "Building with debugging flags")
-    set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -G -Xcompiler -rdynamic")
-endif(CMAKE_BUILD_TYPE MATCHES Debug)
-
-# To apply RUNPATH to transitive dependencies (this is a temporary solution)
-set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--disable-new-dtags")
-set(CMAKE_EXE_LINKER_FLAGS "-Wl,--disable-new-dtags")
-
-# Build options
-option(BUILD_SHARED_LIBS "Build shared libraries" ON)
-option(BUILD_TESTS "Configure CMake to build tests" ON)
-option(BUILD_BENCHMARKS "Configure CMake to build (google) benchmarks" OFF)
-
-###################################################################################################
-# - cudart options --------------------------------------------------------------------------------
 # cudart can be statically linked or dynamically linked. The python ecosystem wants dynamic linking
-
 option(CUDA_STATIC_RUNTIME "Statically link the CUDA runtime" OFF)
 
-if(CUDA_STATIC_RUNTIME)
-    message(STATUS "Enabling static linking of cudart")
-    set(CUDART_LIBRARY "cudart_static")
-else()
-    set(CUDART_LIBRARY "cudart")
-endif(CUDA_STATIC_RUNTIME)
+message(STATUS "Build with NVTX support: ${USE_NVTX}")
+message(STATUS "Configure CMake to build tests: ${BUILD_TESTS}")
+message(STATUS "Configure CMake to build (google) benchmarks: ${BUILD_BENCHMARKS}")
+message(STATUS "Build cuDF shared libraries: ${BUILD_SHARED_LIBS}")
+message(STATUS "Use a file cache for JIT compiled kernels: ${JITIFY_USE_CACHE}")
+message(STATUS "Build and statically link Arrow libraries: ${ARROW_STATIC_LIB}")
+message(STATUS "Use the default allocator for hash tables: ${HT_DEFAULT_ALLOCATOR}")
+message(STATUS "Build with per-thread default stream: ${PER_THREAD_DEFAULT_STREAM}")
+message(STATUS "Auto-detect CUDA architectures to build from the GPUs on the current system: ${AUTO_DETECT_CUDA_ARCHITECTURES}")
+message(STATUS "Disable warnings generated from deprecated declarations: ${DISABLE_DEPRECATION_WARNING}")
+message(STATUS "Enable the -lineinfo option for nvcc (useful for cuda-memcheck / profiler: ${CMAKE_CUDA_LINEINFO}")
+message(STATUS "Statically link the CUDA runtime: ${CUDA_STATIC_RUNTIME}")
 
-###################################################################################################
-# - cmake modules ---------------------------------------------------------------------------------
+# Set a default build type if none was specified
+set(DEFAULT_BUILD_TYPE "Release")
+set(CUDF_BUILD_TESTS ${BUILD_TESTS})
+set(CUDF_BUILD_BENCHMARKS ${BUILD_BENCHMARKS})
 
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/" ${CMAKE_MODULE_PATH})
+set(CUDF_CXX_FLAGS "")
+set(CUDF_CUDA_FLAGS "")
+set(CUDF_CXX_DEFINITIONS "")
+set(CUDF_CUDA_DEFINITIONS "")
```

Comment:
What are these for?

Reply:
These are used in `target_compile_options()` as an alternative to the `ConfigureCuda.cmake` setting the global `CMAKE_CXX_FLAGS`/`CMAKE_CUDA_FLAGS`.

Reply:
Shouldn't we just be setting the flags as part of the target?

Reply:
Yep that's happening here: https://github.com/rapidsai/cudf/blob/db210ffc4b46a775a0ed9a1313b5f0d48bad3704/cpp/CMakeLists.txt#L169-L176